src/components/Select.vue

```diff
@@ -55,7 +55,6 @@
         <li
           role="option"
           v-for="(option, index) in filteredOptions"
-          :key="index"
```

Comment:
Here it is simply using `index` as the key. It is defined but since it is the `index` of the `filteredOptions` it does not stick to the option. This means the same option might get a different `index` when `filteredOptions` changes.

If I understand the purpose of `:key` correctly it should be chosen so that it sticks to the thing you are iterating over, in this case `option`. If you render the same `option`, it should always have the same `:key`, so the dom node will be moved around and must not rerendered completely.

All in all I don't think it makes any real difference here though, because the template is so simple and there are no transitions or other special vue logic here. But if there were, having the `:key=index` might break it.

Most correctly there should be a `:key` for every `option`, but I don't know how to derive it correctly from every `option` format.

Reply:
I think `JSON.stringify` actually might work in 99 % of all cases. Out of my head it will break only in the special case if `option` is an object with a circular reference (you see I assume people pass the weirdest things as option ;-)). Should be tested though.

If this is implemented overrideable as a prop though (e.g. `getOptionId` similar to `getOptionLabel`), then it will work in 99 % of all cases and can be overriden if it fails.

Maybe it should by default look if it is an object and has an `id` property and if yes use it (like with `.label`), otherwise fall back to `JSON.stringify`. Something like

```js
props: {
  getOptionId: {
    type: Function,
    default(option) => {
     if (typeof option === "object" && option.id) {
        return option.id
      } else {
        return JSON.stringify(option)
      }
    }
  }
}
```

Reply:
I added it right now. No tests yet though, do we need them?